examples/websocket_rfq.py

```diff
@@ -6,31 +8,55 @@
 
 logging.getLogger('asyncio').setLevel(logging.WARNING)
 
-import proxy_clients.ws as ws_client
-import protocol.server as server
 
 logging.basicConfig(
     format='%(asctime)s %(levelname)-8s %(message)s',
     level=logging.DEBUG,
     datefmt='%Y-%m-%d %H:%M:%S')
 
 
-async def handle_snapshot(snapshot: server.ServerMessage):
+async def handle_snapshot(snapshot: server.ServerMessage, client: ws_client.ProxyWSClient):
     assert (snapshot.type() == server.MessageType.snapshot)
     logging.info("Snapshot arrived!\n%s", snapshot.body())
 
 
-async def handle_order_message(message: server.ServerMessage):
+async def handle_order_message(message: server.ServerMessage, client: ws_client.ProxyWSClient):
     # Message type one of order_added, order_deleted, order_updated, order_executed
-    logging.info("Order message handled\n%s [%s]", message.type().name, message.body())
+    logging.info(
+        "Order message handled\n%s [%s]", message.type().name, message.body())
+
+
+def legFromString(legString):
+    leg = legString.split("@")
+    return {"sym": leg[0], "ratio": leg[1]}
+
+
+async def handle_order_added_message(message: server.ServerMessage, client: ws_client.ProxyWSClient):
+    body = message.body()
+    logging.info(
+        "Order added message handled\n%s [%s]", message.type().name, body)
+
+    # Check if indicative order
+    if (body["market_id"] == "255"):
+        logging.info("RFQ received for %s %s", body["symbol"], body["side"])
+
+        # Respond to indicative order with firm order
+        side = "buy" if body["side"] == "sell" else "sell"
+        price = "1000.0" if side == "sell" else "900.0"
+        legs = list(map(legFromString, body["symbol"].split("/")))
```

Comment:
Here I am just going to assume that the multileg symbol coming back from the API e.g `BTC-20220930-25000C@1/BTC-20220930-26000C@-1` already has the CPLX format (ascending entity ids and first leg with positive ratio)

Reply:
that's correct assumption, AFAIK; @egor-pt, please confirm here